### ISSUE TITLE
Ghost Monkey and Invis Monkey break the rig

### ISSUE THREAD

**Issue by LazVaz12** [iiDk-the-actual/iis.Stupid.Menu#89]
If you use longarms and extend them all the way and go invis/ghost theres a chance when you go out of it your hand will be stuck extended, the way to fix it is by leaving and rejoining but that is hard to do in some cases, also sometimes other people's rig breaks like that and it makes it confusing as people's arms are extended only for you.

**Comment by Flummidill** [comment]
also happens in guardian mode

**Comment by 64Will64** [comment]
> If you use longarms and extend them all the way and go invis/ghost theres a chance when you go out of it your hand will be stuck extended, the way to fix it is by leaving and rejoining but that is hard to do in some cases, also sometimes other people's rig breaks like that and it makes it confusing as people's arms are extended only for you.

vouch, this happens too, was trying this menu out because of boredom and i believe alot of rig mods just break other people's rigs for some reason.

**Comment by iiDk-the-actual** [comment]
if anyone can figure out how this happens please report back to me, i have no idea

**Comment by Flummidill** [comment]
If you enable always guardian and leave the server while your player is tapping the moon it happens very often.

**Comment by LazVaz12** [comment]
> if anyone can figure out how this happens please report back to me, i have no idea

Use longarms, extend your hand all the way so it's not bent, then it use ghost/invis monkey and it get's stuck. I found a video where you can see it happening on other players (clientsided, your arm being extended is seversided though)
https://www.youtube.com/shorts/Qbb1K8bp_2Y

**Comment by iiDk-the-actual** [comment]
> > if anyone can figure out how this happens please report back to me, i have no idea
> 
> Use longarms, extend your hand all the way so it's not bent, then it use ghost/invis monkey and it get's stuck. I found a video where you can see it happening on other players (clientsided, your arm being extended is seversided though) https://www.youtube.com/shorts/Qbb1K8bp_2Y

Yes, but I need to know why it happens

**Comment by corbin418** [comment]
a quick fix for this would be to enable legacy ghost view.

**Comment by sunny-dee-71** [comment]
> > > if anyone can figure out how this happens please report back to me, i have no idea
> > 
> > 
> > Use longarms, extend your hand all the way so it's not bent, then it use ghost/invis monkey and it get's stuck. I found a video where you can see it happening on other players (clientsided, your arm being extended is seversided though) https://www.youtube.com/shorts/Qbb1K8bp_2Y
> 
> Yes, but I need to know why it happens

I'm not sure but it happens less if you don't use gorillalocomotion.hand thingy and use gorilla tagger.instamce.hand thingy

I am not sure though 
the reason I don't just copy my code is because I'm in school and can't rember the exact names

**Comment by ccatnado** [comment]
Thanks for confirming you aren’t using playerRig.SetActive, that means the issue isn’t from outright disabling the whole rig. But the rig can still break from other things.

	1.	Renderer or Collider Disabling:
	•	If the mod toggles invisibility by disabling Renderer or SkinnedMeshRenderer, and fails to re-enable them correctly, it can break the rig visuals.
	2.	Detaching Parts of the Rig:
	•	Some ghost mods detach or relocate hands/head (setting parent to null, etc.). If not re-parented or positioned back, the rig goes out of sync or floats around.
	3.	Disabling Photon Sync / Gorilla Locomotion Components:
	•	If GorillaTagManager, GorillaLocomotion.Player, or VRRig components are toggled off or messed with, players may go invisible or glitch out across the network.
	4.	Shared State Conflicts:
	•	If both Ghost and Invisible modes modify the same rig components, enabling both at once can result in conflicting states (e.g., ghost disables colliders, invisible disables renderers—re-enabling one might skip restoring the other)

How to fix

To prevent the rig from breaking:
	•	Check what exactly your Ghost and Invisible toggles do.
	•	Look for .enabled = false on:
	•	SkinnedMeshRenderer
	•	BoxCollider, CapsuleCollider, etc.
	•	VRRig references
	•	Create proper restore states for both modes.
	•	When disabling, store original state.
	•	When enabling back, restore exactly what was changed, no more, no less.

**Comment by sunny-dee-71** [comment]
> Thanks for confirming you aren’t using playerRig.SetActive, that means the issue isn’t from outright disabling the whole rig. But the rig can still break from other things.
> 
> ```
> 1.	Renderer or Collider Disabling:
> •	If the mod toggles invisibility by disabling Renderer or SkinnedMeshRenderer, and fails to re-enable them correctly, it can break the rig visuals.
> 2.	Detaching Parts of the Rig:
> •	Some ghost mods detach or relocate hands/head (setting parent to null, etc.). If not re-parented or positioned back, the rig goes out of sync or floats around.
> 3.	Disabling Photon Sync / Gorilla Locomotion Components:
> •	If GorillaTagManager, GorillaLocomotion.Player, or VRRig components are toggled off or messed with, players may go invisible or glitch out across the network.
> 4.	Shared State Conflicts:
> •	If both Ghost and Invisible modes modify the same rig components, enabling both at once can result in conflicting states (e.g., ghost disables colliders, invisible disables renderers—re-enabling one might skip restoring the other)
> ```
> 
> How to fix
> 
> To prevent the rig from breaking: • Check what exactly your Ghost and Invisible toggles do. • Look for .enabled = false on: • SkinnedMeshRenderer • BoxCollider, CapsuleCollider, etc. • VRRig references • Create proper restore states for both modes. • When disabling, store original state. • When enabling back, restore exactly what was changed, no more, no less.

Holy chatgpt

**Comment by ccatnado** [comment]
> > Thanks for confirming you aren’t using playerRig.SetActive, that means the issue isn’t from outright disabling the whole rig. But the rig can still break from other things.
> > ```
> > 1.	Renderer or Collider Disabling:
> > •	If the mod toggles invisibility by disabling Renderer or SkinnedMeshRenderer, and fails to re-enable them correctly, it can break the rig visuals.
> > 2.	Detaching Parts of the Rig:
> > •	Some ghost mods detach or relocate hands/head (setting parent to null, etc.). If not re-parented or positioned back, the rig goes out of sync or floats around.
> > 3.	Disabling Photon Sync / Gorilla Locomotion Components:
> > •	If GorillaTagManager, GorillaLocomotion.Player, or VRRig components are toggled off or messed with, players may go invisible or glitch out across the network.
> > 4.	Shared State Conflicts:
> > •	If both Ghost and Invisible modes modify the same rig components, enabling both at once can result in conflicting states (e.g., ghost disables colliders, invisible disables renderers—re-enabling one might skip restoring the other)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > How to fix
> > To prevent the rig from breaking: • Check what exactly your Ghost and Invisible toggles do. • Look for .enabled = false on: • SkinnedMeshRenderer • BoxCollider, CapsuleCollider, etc. • VRRig references • Create proper restore states for both modes. • When disabling, store original state. • When enabling back, restore exactly what was changed, no more, no less.
> 
> Holy chatgpt

Want me to take it down then? At least I’m helping…

**Comment by ccatnado** [comment]
Hey, I get you didn’t use playerRig.SetActive(false), but something’s still messing with the rig during Ghost or Invisible Monkey. It might be turning off renderers or moving parts of the rig without resetting them after. Even if that line’s not there, the issue could still be coming from other changes to the rig.

Not calling you out or anything, just trying to help figure out what’s going wrong so it can work.

**Comment by sunny-dee-71** [comment]
> > > if anyone can figure out how this happens please report back to me, i have no idea
> > 
> > 
> > Use longarms, extend your hand all the way so it's not bent, then it use ghost/invis monkey and it get's stuck. I found a video where you can see it happening on other players (clientsided, your arm being extended is seversided though) https://www.youtube.com/shorts/Qbb1K8bp_2Y
> 
> Yes, but I need to know why it happens

yeah so i just thought of this but i am pretty sure it is to do with the pole on the GorillaIk getting mesed up

**Comment by RubeGT** [comment]
fart

**Comment by iistupiduser** [comment]
this has been out for a while

**Comment by iiDk-the-actual** [comment]
Fixed in recent commit, will be added next update
